### PR TITLE
Enable various extra extensions for QEMU and GCC

### PR DESCRIPTION
**Pull request by dramforever**
This pull request tries to tackle https://github.com/codewars/runner/issues/218 but I haven't been able to test it. ~~Also not sure if we should add the crypto extensions Zk* in.~~ Since we will have 7.1, added Zk*.

### CODE REVIEW

File: examples/rvv-memcpy/solution.s [hunk 1]
```diff
@@ -0,0 +1,16 @@
+.globl rvv_memcpy
```
Comment (kazk) [inline examples/rvv-memcpy/solution.s:1]: Can you add 

```yaml
 - name: Run rvv-memcpy example 
   run: bin/run rvv-memcpy
```

under

https://github.com/codewars/riscv/blob/0679125cdd8326acf398c617909a681e06191c71/.github/workflows/ci.yml#L45-L46

to run this example in CI?

Comment (dramforever) [reply examples/rvv-memcpy/solution.s:1]: Done. Separate commits for now, will squash if needed after we decide on how to deal with qemu on host.

**Comment by kazk** [comment]
@dramforever You can test against QEMU 7.1 after rebasing (#9).

**Comment by dramforever** [comment]
I ran CI on my own fork and it works fine, both vector and crypto are enabled now, for both GCC and QEMU:

https://github.com/dramforever/codewars-riscv-container/pull/2/checks

File: Dockerfile [hunk 2]
```diff
@@ -30,4 +30,6 @@ WORKDIR /workspace
 COPY workspace/codewars_reporter.c .
 COPY workspace/tests.c .
 
+ENV QEMU_CPU=rv64,v=on,vext_spec=v1.0,zbkx=on,zk=on,zks=on
```
Comment (kazk) [inline Dockerfile]: Let's move this to `bin/run` so the image doesn't require QEMU 7.1 on host by default.

File: bin/run [hunk 3]
```diff
@@ -11,8 +11,10 @@ fi
 
 W=/workspace
 
+ARCH=rv64gcv_zba_zbb_zbc_zbs_zbkx_zk_zks
```
Comment (kazk) [inline bin/run:14]:
```suggestion
ARCH=rv64gcv_zba_zbb_zbc_zbs_zbkx_zk_zks
QEMU_CPU=rv64,v=on,vext_spec=v1.0,zbkx=on,zk=on,zks=on
```

File: bin/run [hunk 4]
```diff
@@ -11,8 +11,10 @@ fi
...
+
 # Create container
-C=$($CONTAINER_ENGINE container create --rm -w $W $IMAGE_TAG sh -c "gcc -O2 solution.s solution_tests.c -lcgreen codewars_reporter.c tests.c -o tests && ./tests")
+C=$($CONTAINER_ENGINE container create --rm -w $W $IMAGE_TAG sh -c "gcc -O2 solution.s -march=$ARCH solution_tests.c -lcgreen codewars_reporter.c tests.c -o tests && ./tests")
```
Comment (kazk) [inline bin/run]:
```suggestion
C=$($CONTAINER_ENGINE container create --rm --env QEMU_CPU=$QEMU_CPU -w $W $IMAGE_TAG sh -c "gcc -O2 solution.s -march=$ARCH solution_tests.c -lcgreen codewars_reporter.c tests.c -o tests && ./tests")
```

File: Dockerfile [hunk 5]
```diff
@@ -30,4 +30,6 @@ WORKDIR /workspace
...
+
```
Comment (kazk) [inline Dockerfile]:
```suggestion
```

**Review by kazk — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (dramforever) [reply Dockerfile]: Done :heavy_check_mark: Thanks for the suggestion

**Review by kazk — APPROVED** [review]

**Comment by StarsDown64** [comment]
As mentioned by myself in https://github.com/codewars/docs/pull/492, I think Zbkx is redundant since it's in both Zk and Zks shorthands. Not a super big change, someone else can prob include in their next test/pr.

**Pull request merged**